katharsis-client/src/main/java/io/katharsis/client/internal/RelationshipRepositoryStubImpl.java

```diff
@@ -84,61 +77,53 @@ private Serializable getSourceId(T source) {
 	@Override
 	public D findOneTarget(I sourceId, String fieldName, QueryParams queryParams) {
 		String url = urlBuilder.buildUrl(sourceClass, sourceId, queryParams, fieldName);
-		BaseResponseContext responseContext = executeGet(url);
-		return (D) responseContext.getResponse().getEntity();
+		return (D) executeGet(url, ResponseType.RESOURCE);
 	}
 
 	@Override
 	public DefaultResourceList<D> findManyTargets(I sourceId, String fieldName, QueryParams queryParams) {
 		String url = urlBuilder.buildUrl(sourceClass, sourceId, queryParams, fieldName);
-		BaseResponseContext responseContext = executeGet(url);
-		return toList(responseContext.getResponse());
+		return (DefaultResourceList<D>) executeGet(url, ResponseType.RESOURCES);
 	}
 
 	@SuppressWarnings("unchecked")
 	@Override
 	public D findOneTarget(I sourceId, String fieldName, QuerySpec querySpec) {
 		String url = urlBuilder.buildUrl(sourceClass, sourceId, querySpec, fieldName);
-		BaseResponseContext responseContext = executeGet(url);
-		return (D) responseContext.getResponse().getEntity();
+		return (D) executeGet(url, ResponseType.RESOURCE);
 	}
 
 	@Override
 	public DefaultResourceList<D> findManyTargets(I sourceId, String fieldName, QuerySpec querySpec) {
 		String url = urlBuilder.buildUrl(sourceClass, sourceId, querySpec, fieldName);
-		BaseResponseContext responseContext = executeGet(url);
-		return toList(responseContext.getResponse());
+		return (DefaultResourceList<D>) executeGet(url, ResponseType.RESOURCES);
 	}
 
 	private void execute(String requestUrl, HttpMethod method, Object targetIds) {
-		JsonPath fieldPath = new ResourcePath(resourceInformation.getResourceType());
+		Document document = new Document();
 
-		JsonApiResponse response = new JsonApiResponse();
-		BaseResponseContext context;
 		if (targetIds instanceof Iterable) {
-			ArrayList<LinkageContainer> containers = new ArrayList<>();
+			ArrayList<ResourceIdentifier> resourceIdentifiers = new ArrayList<>();
 			for (Object targetId : (Iterable<?>) targetIds) {
-				containers.add(new LinkageContainer(targetId, targetClass, relationshipEntry));
+				String strTargetId = resourceInformation.toIdString(targetId);
+				resourceIdentifiers.add(new ResourceIdentifier(strTargetId, resourceInformation.getResourceType()));
 			}
-			response.setEntity(containers);
-			context = new CollectionResponseContext(response, fieldPath, null);
-		}
-		else {
-			Object targetId = targetIds;
-			response.setEntity(new LinkageContainer(targetId, targetClass, relationshipEntry));
-			context = new ResourceResponseContext(response, fieldPath, (QueryAdapter) null);
+			document.setData(Nullable.of((Object) resourceIdentifiers));
+		} else {
+			String strTargetId = resourceInformation.toIdString(targetIds);
```

Comment:
This code is flawed. If it's not `Iterable` then just do this thing. That is literally every other type in the world.